write-fonts/generated/generated_name.rs

```diff
@@ -127,7 +127,7 @@ impl FromObjRef<read_fonts::tables::name::LangTagRecord> for LangTagRecord {
 }
 
 ///[Name Records](https://docs.microsoft.com/en-us/typography/opentype/spec/name#name-records)
-#[derive(Clone, Debug, Default)]
+#[derive(Clone, Debug, Default, PartialEq, Eq, PartialOrd, Ord)]
```

Comment:
The terminology here is exceptionally and arguably unecessarily Rust-focused while the underlying concepts are not. Feels like an unnecessary step backwards from a general schema. For example, why not `#[capabilities(hash, compare, sort)]`

Reply:
^ refers to the `#[derive_traits(...)]` in the codegen inputs, apparently I clicked the wrong file

Reply:
Hmm. I think this is another example of a situation where the annotations in codegen are not always necessarily some language-agnostic thing, but are sometimes very specifically instructions for what code we want to generate for this (rust) project. In this case, the idea that these types need to be (e.g.) hashable isn't really a matter of the spec, but more a consequence of some consuming code somewhere, where I have decided to store something in a hashmap; that may not be the case in another implementation.

Similarly, an attempt to generalize this idea beyond rust is also tricky. Does `compare` mean `PartialEq + Eq`, or will `PartialEq` suffice?

There's a reasonable argument that we should just derive these traits for every type; and the reason I would want to avoid that is just that we have a ton of types, and so deriving a bunch of traits for all of them causes us to generate a bunch of extra code. Maybe this is fine?

Reply:
My naive thought would be capabilities:compare means PartialEq + Eq except where infeasible because the type contains floats.